api/src/main/java/com/openlattice/linking/RealtimeLinkingApi.java

```diff
@@ -32,21 +41,52 @@
  */
 public interface RealtimeLinkingApi {
 
-    String SERVICE               = "/linker";
-    String CONTROLLER            = "/linking";
-    String BASE                  = SERVICE + CONTROLLER;
-
-    String FINISHED              = "/finished";
-    String MATCHED               = "/matched";
-
-    String SET                   = "/set";
-    String LINKING_ID            = "linkingId";
-    String LINKING_ID_PATH       = "/{" + LINKING_ID + "}";
+    String BLOCKING                   = "/blocking";
+    String CONTROLLER                 = "/linking";
+    String FINISHED                   = "/finished";
+    String LINKING_ENTITY_SET_ID      = "linkingEntitySetId";
+    String LINKING_ENTITY_SET_ID_PATH = "/{" + LINKING_ENTITY_SET_ID + "}";
+    String LINKING_ID                 = "linkingId";
+    String LINKING_ID_PATH            = "/{" + LINKING_ID + "}";
+    String LINKS                      = "/links";
+    String MATCHED                    = "/matched";
+    String SET                        = "/set";
 
+    String SERVICE = "/linker";
+    String BASE    = SERVICE + CONTROLLER;
 
     @GET( BASE + FINISHED + SET )
     Set<UUID> getLinkingFinishedEntitySets();
 
     @GET( BASE + MATCHED + LINKING_ID_PATH )
     Set<MatchedEntityPair> getMatchedEntitiesForLinkingId( @Path( LINKING_ID ) UUID linkingId );
+
+    @POST( BASE + BLOCKING )
+    Map<UUID, Map<UUID, List<BlockedEntity>>> block( @Body BlockingRequest blockingRequest );
+
+    UUID createNewLinkedEntity( Set<EntityDataKey> entityDataKeys );
+
+    @GET( BASE + LINKS + LINKING_ENTITY_SET_ID_PATH)
+    Map<UUID, Set<EntityDataKey>> getLinkedEntityKeyIds(
+            @Path( LINKING_ENTITY_SET_ID ) UUID linkingEntitySetId );
+
+    @PUT( BASE + LINKS + LINKING_ENTITY_SET_ID_PATH + LINKING_ID_PATH )
+    Integer setLinkedEntities(
+            @Path( LINKING_ENTITY_SET_ID ) UUID linkingEntitySetId,
+            @Path( LINKING_ID ) UUID linkedEntityKeyId,
+            @Body Set<EntityDataKey> entityDataKeys );
+
+    @POST( BASE + LINKS + LINKING_ENTITY_SET_ID_PATH + LINKING_ID_PATH )
+    Set<EntityDataKey> addLinkedEntities(
+            @Path( LINKING_ENTITY_SET_ID ) UUID linkingEntitySetId,
+            @Path( LINKING_ID ) UUID linkedEntityKeyId,
+            @Body Set<EntityDataKey> entityDataKeys );
+
+    @HTTP( method = "DELETE",
```

Comment:
is there a reason this isn't a `@DELETE( BASE + ... )` ?

Reply:
Needs to have a body.

Reply:
ahh makes sense